App.js

```diff
@@ -1,26 +1,248 @@
+      response => response.json(),
+    )   
+  }
+  
+  const randomRangeNumber = (max) => {
+    return Math.floor(Math.random() * max)
+  }
+  
+  const getRandomCharacter = (characters) => {
+    let character = characters[randomRangeNumber(characters.length) - 1]
+  
+    while (!character.house) {
+      character = characters[randomRangeNumber(characters.length) - 1]
+    }
+    return character 
+  }
+  
+  const getRandomCharacterRecursion = (characters) => {
+    let character = characters[randomRangeNumber(characters.length) - 1]
+  
+    // stop condition
+    if (!character.house) {
+      return getRandomCharacterRecursion(characters)
+    }
+  
+    return character
+  }
+
+  // Run houseMatch to see if guess is correct
+  const houseMatch = (house, character) => {
+    console.log(character);
+    if (house === character.house) {
+       return "Correct!";
+    } else {
+      return `No, ${character.name} is in ${character.house}`;
+    }
+  };
+
+  
+  
+  export default function App() {
+    const [character, setCharacter] = useState()
+    const [house, setHouse] = useState()
+    const [answer, setAnswer] = useState(false)
+    const [DisplayGame, setDisplayGame] = useState (false)
+    const [DisplayStart, setDisplayStart] = useState (true)
+    // const [DisplayCharacter, setDisplayCharacter] = useState (false)
+    
+
+    
+  
+    const handleFetchCharacters = () => {
+      fetchCharacters().then(json => {
+        // recursion
+        // let character = getRandomCharacterRecursion(json)
+        // no recursion
+        let character = getRandomCharacter(json)
+        setCharacter(character)
+      })
+    }
+
+    const houseMatch = (house) => {
+      if (house === character.house) {
+        return (`Correct! ${character.name} is in ${character.house} at ${character.school}`)
+      } else {
+        return (`No, ${character.name} is in ${character.house}`)
+      }
+    }
+  
+    useEffect(() => {
+      handleFetchCharacters()
+    }, [])
+  
+    if (!character) {
+      return null
+    }
+  
+    return (
+      <MainContain>
+        {/* START GAME CONTAINER + STATE */}
+        {DisplayStart === false || answer === true ? (null) : (
+       <Container>
+         <Header>Are you a Potterhead?</Header>
+         <Buttonz
+        title="Find out!"
+        onPress={() => {
+          handleFetchCharacters()
+          setDisplayStart(false)
+          setDisplayGame(true)
+        }}>
+          <Writing>Find out!</Writing>
+      </Buttonz>
+       </Container>
+       )}
+
+       {/* GAME COMP */}
 
-export default App
+        {DisplayGame === false ? (null) : (
+          <Container>
+            <Writing>In which house does</Writing>
+          <Header>
+            {character.name}
+          </Header>
+          <Writing>belong?</Writing>
+         <HouseContainer>
+         <Buttonz
+              title="Gryffindor"
+              onPress={() => {
```

Comment:
Here you could have an array of all the houses
`const houses = ['Gryffindor', 'Slytherin', ...]`
and map through that array to render the buttonz component for each one of them
```
houses.map(house => (
    <Buttonz
        title={house}
        onPress={() => {
           setAnswer(houseMatch(house, character));
           setDisplayGame(false)
      }}>
         <Writing>{house}</Writing>
    </Buttonz>
))
```
To save some time and keep the code DRY